app/src/main/java/com/example/soptseminar1/HomeActivity.kt

```diff
@@ -2,51 +2,59 @@ package com.example.soptseminar1
 
 import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
+import androidx.viewpager2.widget.ViewPager2
 import com.example.soptseminar1.databinding.ActivityHomeBinding
 
 class HomeActivity : AppCompatActivity() {
 
-    private var position = FOLLOWER
     private lateinit var binding: ActivityHomeBinding
+    private lateinit var mainViewPagerAdapter: MainViewPagerAdapter
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         binding = ActivityHomeBinding.inflate(layoutInflater)
         setContentView(binding.root)
 
-        initTransactionEvent()
+        initAdapter()
+        initBottomNavi()
     }
 
-    private fun initTransactionEvent(){
-        val fragment1 = FollowerFragment()
-        val fragment2 = RepositoryFragment()
+    private fun initAdapter(){
+        val fragmentList = listOf(ProfileFragment(), HomeFragment(), CameraFragment())
+        mainViewPagerAdapter = MainViewPagerAdapter(this)
+        mainViewPagerAdapter.fragments.addAll(fragmentList)
 
-        supportFragmentManager.beginTransaction().add(R.id.main_fragment, fragment1).commit()
+        binding.vpMain.adapter = mainViewPagerAdapter
+    }
 
-        binding.btnFollower.setOnClickListener{
-            val transaction = supportFragmentManager.beginTransaction()
-            when (position) {
-                REPOSITORY -> {
-                    transaction.replace(R.id.main_fragment, fragment1)
-                    position = FOLLOWER
-                }
+    private fun initBottomNavi(){
+        binding.vpMain.registerOnPageChangeCallback(object : ViewPager2.OnPageChangeCallback(){
+            override fun onPageSelected(position: Int) {
+                binding.bnvMain.menu.getItem(position).isChecked = true
             }
-            transaction.commit()
-        }
+        })
 
-        binding.btnRepo.setOnClickListener{
-            val transaction = supportFragmentManager.beginTransaction()
-            when (position) {
-                FOLLOWER -> {
-                    transaction.replace(R.id.main_fragment, fragment2)
-                    position = REPOSITORY
+        binding.bnvMain.setOnItemSelectedListener {
+            when(it.itemId){
+                R.id.menu_profile -> {
+                    binding.vpMain.currentItem = FIRST_FRAGMENT
+                    return@setOnItemSelectedListener true
+                }
+                R.id.menu_home -> {
+                    binding.vpMain.currentItem = SECOND_FRAGMENT
+                    return@setOnItemSelectedListener true
+                }
+                else -> {
+                    binding.vpMain.currentItem = THIRD_FRAGMENT
+                    return@setOnItemSelectedListener true
                 }
```

Comment:
```suggestion
return@setOnItemSelectedListener when(it.itemId){
                R.id.menu_profile -> {
                    binding.vpMain.currentItem = FIRST_FRAGMENT
                    true
                }
                R.id.menu_home -> {
                    binding.vpMain.currentItem = SECOND_FRAGMENT
                    true
                }
                else -> {
                    binding.vpMain.currentItem = THIRD_FRAGMENT
                    true
                }
            }
```
이렇게 쓸 수도 있을 것 같습니다. 저는 Java보다는 C와 C++에 더 익숙한 편인데, 기존 C나 C++에서는 조건문 안에서 매번 return을 써줘야 했습니다. 하지만 Kotlin에서는 조건문 앞에 일단 먼저 `return`을 작성하고, 각각의 분기문 안의 맨 마지막에 반환값을 툭! 던져주기만 해도 잘 굴러갑니다. 그 이유는, Kotlin에서는 `if`나 `when`이 값을 리턴하기 때문입니다.